week3_cardgame/cardgame/src/components/card/CardHandler.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect, useState } from 'react';
+import Card from './Card';
+import CARDLIST from './CardData';
+import styled from 'styled-components';
+
+export default function CardHandler({ level, updateScore, shuffle }) {
+    const [cards, setCards] = useState([]);
+    const [selectedCard, setSelectedCard] = useState(null); // 전에 선택한 카드
+    const [matchCards, setMatchCards] = useState([]); // 매칭된 카드들
+    const [openCards, setOpenCards] = useState([]); // 오픈된 카드들 (선택한 카드 + 매칭된 카드들)
+
+    // 랜덤 카드 뽑기
+    const getRandomElement = (arr) => {
+        console.log('getRandomElement');
+        let result = new Set();
+        while (result.size < level) {
+            const randomElement = arr[Math.floor(Math.random() * arr.length)];
+            result.add(randomElement);
+        }
+        return Array.from(result);
+    };
+
+    // 카드 쌍 만들기
+    const initialCards = () => {
+        const selectedCards = getRandomElement(CARDLIST);
+        const cardPairs = selectedCards.reduce(
+            (acc, card) => [
+                ...acc,
+                { ...card, id: card.id + 1000 }, // ID 충돌 방지
+                { ...card, id: card.id + 2000 }, // ID 충돌 방지
+            ],
+            [],
+        );
+        return cardPairs.sort(() => 0.5 - Math.random());
+    };
+
+    useEffect(() => {
+        setCards(initialCards());
+        setSelectedCard(null);
+        setMatchCards([]);
+        setOpenCards([]);
+    }, [level, shuffle]);
+
+    function chooseCard(e, id) {
+        e.stopPropagation();
+        // 매칭된 카드와 자기 자신은 선택 불가
+        if (!matchCards.includes(id) && id !== selectedCard) {
+            setOpenCards((prev) => [...prev, id]); // 카드 오픈
+            if (selectedCard == null) {
```

Comment:
null은 falsy한 값이니 이렇게 써줄수도 있겠네요!!

```suggestion
            if (!selectedCard) {
```

물론 selectedCard가 null인 경우와 다른 falsy한 값(0 등)일 경우가 다른 경우로 분기되어야 하는 상황이라면 채현이가 작성한 대로 하는 것이 맞습니다!!

Reply:
이렇게 쓰니 훨씬 코드가 간결해지네요! 감사합니다 수정하겠습니다👍🏻👍🏻